pkg/network/ebpf/c/protocols/postgres/decoding.h

```diff
@@ -107,12 +108,28 @@ static int __always_inline skip_string(pktbuf_t pkt, int message_len) {
     return SKIP_STRING_FAILED;
 }
 
-// Main processing logic for the Postgres protocol. It reads the first message header and decides what to do based on the
-// message tag. If the message is a new query, it stores the query in the in-flight map. If the message is a command
-// complete, it enqueues the transaction and deletes it from the in-flight map. If the message is not a command complete,
-// it tries to read up to POSTGRES_MAX_MESSAGES messages, looking for a command complete message.
-// If the message is not a new query or a command complete, it ignores the message.
-static __always_inline void postgres_entrypoint(pktbuf_t pkt, conn_tuple_t *conn_tuple, struct pg_message_header *header, __u8 tags) {
+// postgres_handle_message reads the first message header and decides what to do based on the
+// message tag. If the message is a new query, it stores the query in the in-flight map.
+// If the message is a parse message, we tail call to the dedicated process_parse_message program.
+// If the message is a command complete, it calls the handle_command_complete program.
+static __always_inline void postgres_handle_message(pktbuf_t pkt, conn_tuple_t *conn_tuple, struct pg_message_header *header, __u8 tags) {
+    const __u32 zero = 0;
+    // If the message is a parse message, we tail call to the dedicated function to handle it.
+    if (header->message_tag == POSTGRES_PARSE_MAGIC_BYTE) {
+        pktbuf_tail_call_option_t process_parse_tail_call_array[] = {
+                [PKTBUF_SKB] = {
+                    .prog_array_map = &protocols_progs,
+                    .index = PROG_POSTGRES_PROCESS_PARSE_MESSAGE,
+                },
+                [PKTBUF_TLS] = {
+                    .prog_array_map = &tls_process_progs,
+                    .index = PROG_POSTGRES_PROCESS_PARSE_MESSAGE,
+                },
+            };
+        pktbuf_tail_call_compact(pkt, process_parse_tail_call_array);
+        return;
+    }
```

Comment:
why do we need the dedicated tail call for parse message?

Reply:
put a documentation that states the parse processor is very complicated, and must be processed in a dedicated tail call